packages/assets-controllers/src/NftController.ts

```diff
@@ -222,41 +218,235 @@ const controllerName = 'NftController';
+    getERC721OwnerOf: AssetsContractController['getERC721OwnerOf'];
+    getERC1155BalanceOf: AssetsContractController['getERC1155BalanceOf'];
+    getERC1155TokenURI: AssetsContractController['getERC1155TokenURI'];
+    onNftAdded?: (data: {
+      address: string;
+      symbol: string | undefined;
+      tokenId: string;
+      standard: string | null;
+      source: string;
+    }) => void;
+    messenger: NftControllerMessenger;
+    config?: Partial<NftConfig>;
+    state?: Partial<NftControllerState>;
+  }) {
+    super({
+      name: controllerName,
+      metadata,
+      messenger,
+      state: {
+        ...defaultNftControllerState,
+        ...state,
+      },
+    });
+    this.#config = {
+      selectedAddress: '',
+      chainId: initialChainId,
+      ipfsGateway: IPFS_DEFAULT_GATEWAY_URL,
+      openSeaEnabled: false,
+      useIpfsSubdomains: true,
+      isIpfsGatewayEnabled: true,
+      ...config,
+    };
+
+    this.#getERC721AssetName = getERC721AssetName;
+    this.#getERC721AssetSymbol = getERC721AssetSymbol;
+    this.#getERC721TokenURI = getERC721TokenURI;
+    this.#getERC721OwnerOf = getERC721OwnerOf;
+    this.#getERC1155BalanceOf = getERC1155BalanceOf;
+    this.#getERC1155TokenURI = getERC1155TokenURI;
+    this.#onNftAdded = onNftAdded;
+
+    this.messagingSystem.subscribe(
+      'PreferencesController:stateChange',
+      this.#onPreferencesControllerStateChange.bind(this),
+    );
+
+    this.messagingSystem.subscribe(
+      'NetworkController:networkDidChange',
+      this.#onNetworkControllerNetworkDidChange.bind(this),
+    );
+  }
+
+  /**
+   * Handles the network change on the network controller.
+   * @param networkState - The new state of the preference controller.
+   * @param networkState.selectedNetworkClientId - The current selected network client id.
+   */
+  #onNetworkControllerNetworkDidChange({
+    selectedNetworkClientId,
+  }: NetworkState) {
+    const {
+      configuration: { chainId },
+    } = this.messagingSystem.call(
+      'NetworkController:getNetworkClientById',
+      selectedNetworkClientId,
+    );
+    this.#config.chainId = chainId;
+  }
+
+  /**
+   * Handles the state change of the preference controller.
+   * @param preferencesState - The new state of the preference controller.
+   * @param preferencesState.selectedAddress - The current selected address.
+   * @param preferencesState.ipfsGateway - The configured IPFS gateway.
+   * @param preferencesState.openSeaEnabled - Controls whether the OpenSea API is used.
+   * @param preferencesState.isIpfsGatewayEnabled - Controls whether IPFS is enabled or not.
+   */
+  async #onPreferencesControllerStateChange({
+    selectedAddress,
+    ipfsGateway,
+    openSeaEnabled,
+    isIpfsGatewayEnabled,
+  }: PreferencesState) {
+    this.#config = {
+      ...this.#config,
+      selectedAddress,
+      ipfsGateway,
+      openSeaEnabled,
+      isIpfsGatewayEnabled,
+    };
+
+    const needsUpdateNftMetadata =
```

Comment:
This seems like a good place to use the `setConfig` method.

```suggestion
    this.setConfig({
      selectedAddress,
      ipfsGateway,
      openSeaEnabled,
      isIpfsGatewayEnabled,
    });

```

@mcmire Maybe we should port `configure` and related methods from `BaseControllerV1` to `BaseController` if we want to use config objects in V2 controllers. This method seems like a redundant solution for a solved problem.

Reply:
For other conversions, we have not been porting `configure` to BaseController v2 controllers. Rather, anything that was previously in `config`, we have merged into the `options`, and we have stored them inside of the controller as private properties. Can we do the same for this controller?

Reply:
I agree. It doesn't seem like we need a config object, when the config options are rarely if ever updated together. We should be able to handle them independently as class fields.